### PR TITLE
Optimize large download, update docs

### LINKED ISSUE #11: Download of large instances needs optimization
Currently, simexpal downloads entire files to RAM (in `instances.py`) and afterwards writes them to disk. Rewrite this to iteratively write smaller chunks to disk, or use something like the `sendfile()` system call.

### PR DESCRIPTION
**Pull request by duylethanh**
This PR contains the following:
- fix #11 
- updated introduction for ``docs/instances``

### CODE REVIEW

File: simexpal/instances.py [hunk 1]
```diff
@@ -22,6 +22,12 @@ class DownloadException(Exception):
 def download_instance(inst_yml, instances_dir, filename, partial_path, ext):
 	import requests
 
+	def download_as_stream(url, download_path):
+		with requests.get(url, stream=True) as request:
+			with open(download_path, 'wb') as f:
+				for chunk in request.iter_content(chunk_size=4096):
```
Comment (avdgrinten) [inline simexpal/instances.py:28]: 4096 is rather small. I think it would be fine to reserve 64KiB or even a megabyte or so.

Comment (duylethanh) [reply simexpal/instances.py]: I changed it to 1MB (``chunk_size=1000000``)

**Comment by duylethanh** [comment]
PR updated.

**Comment by avdgrinten** [comment]
LGTM.

**Pull request merged**